### PR TITLE
HeavyIon Jet DQM

### PR DESCRIPTION
**Pull request by BetterWang**
DQM for HeavyIon jet, supersede #7405  from a clean branch.

### CODE REVIEW

**Comment by deguio** [comment]
please test

File: DQMOffline/JetMET/python/jetAnalyzer_cff.py [hunk 1]
```diff
@@ -15,3 +15,34 @@
 
 
 jetDQMAnalyzerSequenceMiniAOD = cms.Sequence(jetDQMAnalyzerAk4PFCHSUncleanedMiniAOD*jetDQMAnalyzerAk4PFCHSCleanedMiniAOD)
+
+jetDQMAnalyzerSequenceHI = cms.Sequence(jetDQMAnalyzerIC5CaloHIUncleaned
```
Comment (deguio) [inline DQMOffline/JetMET/python/jetAnalyzer_cff.py]: is this sequence actually included in any central sequence?

File: Validation/RecoJets/python/JetValidationHeavyIons_cff.py [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+#                                    srcRho         = cms.InputTag("rho", "ak4PFJets"),
+                                    srcGen         = cms.InputTag("ak5HiCleanedGenJets"),
+                                    JetCorrections = cms.string(""),
+                                    recoJetPtThreshold = cms.double(40),
+                                    genEnergyFractionThreshold     = cms.double(0.05),
+                                    matchGenPtThreshold                 = cms.double(20.0),
+                                    RThreshold                     = cms.double(0.3)
+                                    )
+'''
+hiJetValidation = cms.Sequence(
```
Comment (deguio) [inline Validation/RecoJets/python/JetValidationHeavyIons_cff.py]: is this sequence actually included in any central sequence?

**Comment by rkunnawa** [comment]
Hi @deguio im not sure what you mean by central sequence, but the HI sequences are included here https://github.com/rkunnawa/cmssw/blob/JetDQM_HeavyIons/DQMOffline/JetMET/python/jetMETDQMOfflineSourceHI_cff.py#L8 which is in turn available in the https://github.com/rkunnawa/cmssw/blob/JetDQM_HeavyIons/DQMOffline/Configuration/python/DQMOfflineHeavyIons_cff.py#L69
similarly for the validation, the sequence is included here https://github.com/rkunnawa/cmssw/blob/JetDQM_HeavyIons/Validation/Configuration/python/ValidationHI_cff.py#L14 and defined in the https://github.com/rkunnawa/cmssw/blob/JetDQM_HeavyIons/Validation/RecoHI/python/globalValidationHeavyIons_cff.py#L51. 
so when we run the workflow 140.53 or 140.0 these get called and the following jets defined in them are run. hope this answers your question.

File: DQMOffline/JetMET/python/jetAnalyzer_cff.py [hunk 3]
```diff
@@ -15,3 +15,34 @@
```
Comment (BetterWang) [inline DQMOffline/JetMET/python/jetAnalyzer_cff.py]: L14 can actually be removed.

**Comment by deguio** [comment]
mmm
I miss something here.
as @BetterWang commented the DQMOffline sequence is defined twice.
same for the validation one. A JetValidationHeavyIons_cff.py already exist in Validation/RecoHI and I see that you added it to Validation/RecoJets as well.

not sure I get the logic. could you please clarify?
thanks,
F.

**Comment by rkunnawa** [comment]
the validation sequence defined in Validation/RecoHI is not used (its the default one that was present before i started working on this). the sequence that runs, which is defined in Validation/RecoHI/python/globalValidationHeavyIons_cff.py#L5 calls from the Validation/RecoJets/JetValidationHeavyIons_cff. i guess one reason why i wrote it that way is that i wanted to have the both the PP jet and heavyion modules to be located in the same place for easier readability. for instance we can use the pp modules to read HI reconstructed non background subttracted jets instead of rewriting a separate module for that. hope this is clear

**Comment by rkunnawa** [comment]
if its necessary for those sequences (which are not used) to be deleted I can remove them quickly and push another commit. please advise.

**Comment by deguio** [comment]
please remove whatever is not used (sequences and files).
thanks,
F.

**Comment by rkunnawa** [comment]
removed sequences and files which are not used.

**Comment by deguio** [comment]
+1

**Comment by davidlange6** [comment]
+1

File: DQMOffline/JetMET/src/JetAnalyzer_HeavyIons.cc [hunk 4]
```diff
@@ -0,0 +1,1141 @@
...
+  Float_t psin_value[200];
+
+  Float_t vn[fourierOrder_][etaBins_];
+  Float_t psin[fourierOrder_][etaBins_];
+  Float_t sumpT[etaBins_];
+
+  double edge_pseudorapidity[etaBins_ +1] = {-5.191, -2.650, -2.043, -1.740, -1.479, -1.131, -0.783, -0.522, 0.522, 0.783, 1.131, 1.479, 1.740, 2.043, 2.650, 5.191 };
+  
+  UEParameters vnUE(vn_.product(),fourierOrder_,etaBins_);
+  const std::vector<float>& vue = vnUE.get_raw();
```
Comment (ktf) [inline DQMOffline/JetMET/src/JetAnalyzer_HeavyIons.cc]: This is breaking the gcc481 IB due to the fact it's unused. Can you please fix it ASAP? @deguio

Comment (deguio) [inline DQMOffline/JetMET/src/JetAnalyzer_HeavyIons.cc]: give me one min

Comment (deguio) [inline DQMOffline/JetMET/src/JetAnalyzer_HeavyIons.cc]: #7588 should do the job.
ciao,
F.

Comment (davidlange6) [inline DQMOffline/JetMET/src/JetAnalyzer_HeavyIons.cc]: ciao @ktf - why do we see this problem only gcc481? do you know?

Comment (ktf) [inline DQMOffline/JetMET/src/JetAnalyzer_HeavyIons.cc]: Yes. This kind of issue is optimization specific, i.e. it depends on when the compiler decides the variable is being unused. I've seen it many times in the past.

**Pull request merged**